draft-ietf-quic-transport.md

```diff
@@ -1453,20 +1421,124 @@ of three ways:
    enabled when it is expensive to send an explicit close, such as mobile
    networks that must wake up the radio.
 
-3. Abrupt Shutdown: An endpoint may send a Public Reset packet at any time
-   during the connection to abruptly terminate an active connection.  A Public
-   Reset packet SHOULD only be used as a final recourse.  Commonly, a public
-   reset is expected to be sent when a packet on an established connection is
-   received by an endpoint that is unable decrypt the packet.  For instance, if
-   a server reboots mid-connection and loses any cryptographic state associated
-   with open connections, and then receives a packet on an open connection, it
-   should send a Public Reset packet in return.  (TODO: articulate rules around
-   when a public reset should be sent.)
+3. Stateless Reset: An endpoint that loses state can use this procedure to cause
+   the connection to terminate early, see {{stateless-reset}} for details.
 
 TODO: Connections that are terminated are added to a TIME_WAIT list at the
 server, so as to absorb any straggler packets in the network.  Discuss TIME_WAIT
 list.
 
+
+## Stateless Reset {#stateless-reset}
+
+A stateless reset is provided as an option of last resort for a server that does
+not have access to the state of a connection.  A server crash or outage might
+result in clients continuing to send data to a server that is unable to properly
+continue the connection.  A server that wishes to communicate a fatal connection
+error MUST use a CONNECTION_CLOSE frame if it has sufficient state to do so.
+
+To support this process, the server sends a stateless_reset_token value during
+the handshake in the transport parameters.  This value is protected by
+encryption, so only client and server know this value.
+
+A server that receives packets that it cannot process sends a packet in the
+following layout:
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+
+|0|C|K|  00001  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Comment:
Should CK be required to be 10 here? Keyphase makes little sense, and Connection ID is required.